runner/android_junit_runner/java/androidx/test/internal/runner/junit4/statement/UiThreadStatement.java

```diff
@@ -88,7 +88,9 @@ public static boolean shouldRunOnUiThread(FrameworkMethod method) {
 
   private static boolean hasAnnotation(
       FrameworkMethod method, Class<? extends Annotation> annotationClass) {
-    return annotationClass != null && method.getAnnotation(annotationClass) != null;
+    return annotationClass != null
+        && (method.getAnnotation(annotationClass) != null
+            || method.getDeclaringClass().isAnnotationPresent(annotationClass));
```

Comment:
@brettchabot What about superclasses/interfaces, shouldn't those be handled?

Reply:
Fair point. I'll drum up a follow-up to handle superclasses/interfaces

Reply:
https://github.com/android/android-test/pull/805 👍